src/universal/modules/meeting/helpers/handleRedirects.js

```diff
@@ -29,15 +30,31 @@ export default function handleRedirects(oldProps, nextProps) {
     const {countName, arrayName} = localPhaseInfo.items;
     const initialPhaseItemCount = nextProps[countName];
     const phaseItems = nextProps[arrayName];
+
     // bail out fast if the query or sub items haven't returned
     if (initialPhaseItemCount === null || initialPhaseItemCount > phaseItems.length) {
-      return false;
+      return undefined;
     }
 
     // if it's a bad number (or not a number at all)
     if (localPhaseItem > phaseItems.length || localPhaseItem <= 0) {
+      // did an item get removed?
+      const oldPhaseItems = oldProps[arrayName];
+      if (oldPhaseItems.length > phaseItems.length) {
```

Comment:
there is a LOT of business logic in there. i keep thinking there's gotta be a way to greatly simplify it, but every time i dig in i just recreate the same thing.